cmd/traffic/cmd/manager/internal/mutator/agent_injector.go

```diff
@@ -274,13 +280,55 @@ func addAgentContainer(
 			appContainer,
 			containerPort,
 			int(appPort.ContainerPort),
+			int(env.APIPort),
 			env.ManagerNamespace,
 			setGID,
 		)})
 
 	return patches, nil
 }
 
+// addTPEnv adds telepresence specific environment variables to the app container
+func addTPEnv(pod *corev1.Pod, cn *corev1.Container, env map[string]string, patches []patchOperation) []patchOperation {
+	if len(env) == 0 {
+		return patches
+	}
+	cns := pod.Spec.Containers
+	var containerPath string
+	for i := range cns {
+		if &cns[i] == cn {
+			containerPath = fmt.Sprintf("/spec/containers/%d", i)
+			break
+		}
+	}
+	keys := make([]string, len(env))
+	i := 0
+	for k := range env {
+		keys[i] = k
+		i++
+	}
+	sort.Strings(keys)
+	if cn.Env == nil {
+		patches = append(patches, patchOperation{
+			Op:    "replace",
```

Comment:
In my mind, this is an assignment, and assigning is always "replace", whereas an "add" is adding to a collection. Consider this in go.
```go
foo = 1
```
here, we replaced the value in the variable `foo` with `1`. One could argue that:
```go
foo := 1
```
is an add, because here, `foo` is also declared. So the question is, do we treat the patch as something that declares the `Env` or assigns a value to something that already exists in the container? In my mind, the `Container` is a struct, and the `Env` is a field in that struct. Since we don't declare it, we don't add it. We replace the value of an existing field. Consequently, the `initiContainers` should probably change from "add" to "replace".